### DISCUSSION TITLE
Destination Airport codes

### DISCUSSION THREAD

**Discussion by gvanem** [szpajder/dumpvdl2#33] [General]
Hello folks. 

I've built `dumpvdl2.exe` on Windows using clang-cl. It works very well using a `--rtlsdr`  or a `--sdrplay` device.
Example message:
```
[2023-01-17 14:28:03 ] [136.975] [-18.7/-45.4 dBFS] [26.7 dB] [1.3 ppm]
479301 (reserved, Airborne) -> 29D8F7 (reserved): Command
AVLC type: U (XID) P/F: 1
 XID: Link Establishment
  Public params:
   Parameter set ID: 8885:1993
   Procedure classes: 00 01
   HDLC options: 20 a4 88
  VDL params:
   Parameter set ID: V
   XID sequencing: seq: 0 retry: 0
   Modulation support: VDL-M2, D8PSK, 31500 bps
   Destination airport: ENZV
   Aircraft location: 60.2N 5.1E 0 ft
   AVLC specific options: 24
   Connection management: 00
```
Now I'd like to add similar ACARS decoding to my own [ADSB decoder](https://github.com/gvanem/Dump1090/) (using a 2nd rtlsdr/sdrplay device) and show additional information like **Destination Airport:  ENZV** shown above. So where do I get a list of such codes?

> `479301 (reserved, Airborne) -> 29D8F7 (reserved): Command`

I assume the code `29D8F7` is the ground-station code. How to get a clear-text of these?

**Comment by olyair11** [comment]
Ground Stations details feature is supported since v1.8.0
https://github.com/szpajder/dumpvdl2/releases/tag/v1.8.0

**Comment by szpajder** [comment]
> I've built dumpvdl2.exe on Windows using clang-cl. It works very well using a --rtlsdr or a --sdrplay device.
> Example message:
> 
> [2023-01-17 14:28:03 ] [136.975] [-18.7/-45.4 dBFS] [26.7 dB] [1.3 ppm]
> 479301 (reserved, Airborne) -> 29D8F7 (reserved): Command

Something is not right with this build - 479301 is an aircraft address and it should be identified as "Aircraft" rather than "reserved". Similarly, 29D8F7 should be identified as "Ground station", not "reserved".

> I assume the code 29D8F7 is the ground-station code. How to get a clear-text of these?

Subscribe to the [acars-vdl2 mailing list](https://acars-vdl2.groups.io). Then grab the [VDL2_Ground_Stations.txt](https://acars-vdl2.groups.io/g/main/files/VDL2_Ground_Stations.txt) file from the Files section. This is a community-made collection of mappings between GS addresses and their locations, so it's by no means complete, but it gets updated frequently.

Also for SITA ground stations (ie. the ones with addresses starting with "2") the 3-letter airport code is embedded in the address itself. See [this post](https://acars-vdl2.groups.io/g/main/message/248) for details. This scheme does not apply to ARINC stations though.

**Comment by gvanem** [comment]
> Something is not right with this build - 479301 is an aircraft address and it should be identified as "Aircraft" rather than "reserved". 
> Similarly, 29D8F7 should be identified as "Ground station", not "reserved".

I assume that's this array:
https://github.com/szpajder/dumpvdl2/blob/f671d9ecf8d4fc60190c097daaad5df9404622b2/src/avlc.c#L127-L136
